components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -134,34 +134,37 @@ export class PrebuildManager {
                     `Running prebuilds without a project is no longer supported. Please add '${cloneURL}' as a project in a Gitpod team.`,
                 );
             }
-            const existingPB = await this.findNonFailedPrebuiltWorkspace({ span }, cloneURL, commitSHAIdentifier);
 
-            // If the existing prebuild is failed, it will be retriggered in the afterwards
             const config = await this.fetchConfig({ span }, user, context);
-            if (existingPB) {
-                // If the existing prebuild is based on an outdated project config, we also want to retrigger it.
-                const existingPBWS = await this.workspaceDB.trace({ span }).findById(existingPB.buildWorkspaceId);
-                const existingConfig = existingPBWS?.config;
-                log.debug(
-                    `startPrebuild | commits: ${commitSHAIdentifier}, existingPB: ${
-                        existingPB.id
-                    }, existingConfig: ${JSON.stringify(existingConfig)}, newConfig: ${JSON.stringify(config)}}`,
-                );
-                const filterPrebuildTasks = (tasks: TaskConfig[] = []) =>
-                    tasks
-                        .map((task) =>
-                            Object.keys(task)
-                                .filter((key) => ["before", "init", "prebuild"].includes(key))
-                                // @ts-ignore
-                                .reduce((obj, key) => ({ ...obj, [key]: task[key] }), {}),
-                        )
-                        .filter((task) => Object.keys(task).length > 0);
-                const isSameConfig =
-                    JSON.stringify(filterPrebuildTasks(existingConfig?.tasks)) ===
-                    JSON.stringify(filterPrebuildTasks(config?.tasks));
-                // If there is an existing prebuild that isn't failed and it's based on the current config, we return it here instead of triggering a new prebuild.
-                if (isSameConfig) {
-                    return { prebuildId: existingPB.id, wsid: existingPB.buildWorkspaceId, done: true };
+
+            if (!forcePrebuild) {
```

Comment:
I looks like `!forcePrebuild` is always `true` for manually triggered prebuilds, otherwise (on push events) it's `false`. 🤔 

Such changes to `startPrebuild` are hard to review, as we need to consider all call sites. Is there a way to improve readability of this part in this PR? Asking, because though I think this makes sense and should do what it advertises, it took way to long to get the context from call sites. 

Would you mind adding some comments at least?

Reply:
Thanks for your comment @AlexTugarev ! Happy to add more comments if that's helpful.

If you open Gitpod in Gitpod, you can easily see all references of `startPrebuild` -- there you'll see that:
- all context parsers (e.g. GitHub, GitLab, Bitbucket, etc) call `startPrebuild` without specifying `forcePrebuild` (i.e. it's `false`)
- the only manual trigger (i.e. server.triggerPrebuild, called from the dashboard) always calls it with `forcePrebuild: true`

So, `forcePrebuild` could also be renamed to something like "manually triggered".

Now, what sort of comment/explanation would you find helpful here? 👀